### ISSUE TITLE
[TT-1983] - `api_id` not used when creating/updating API definition

### ISSUE THREAD

**Issue by tomirio619** [TykTechnologies/tyk#2979]
**Branch/Environment/Version**
- Branch/Version: 2.9.3.1
- Environment: Hybrid

**Describe the bug**
I'm experiencing the same problem as described in #1265.
Creating an API with POST on `/api/apis` overrides the provided `api_id`. 
A PUT on that same API `/api/apis/id` with an `api_id` in the API definition does not change the `api_id`.

**Reproduction steps**
An example that shows this behavior:
```python
import json
import os

import requests

TYK_DASHBOARD_API_URL = os.getenv("TYK_DASHBOARD_API_URL")
TYK_DASHBOARD_API_KEY = os.getenv("TYK_DASHBOARD_API_KEY")

session = requests.Session()
session.headers.update({"Authorization": TYK_DASHBOARD_API_KEY})

api_def = """
{
  "api_id": "e830046147ae4a7d7c98570783ce67ff",
  "org_id": "5e554e19274a9600012a3eb1",
  "name": "My Example API",
  "slug": "my-example-apii",
  "active": true,
  "auth": {
    "auth_header_name": "authorization"
  },
  "definition": {
    "location": "header",
    "key": "version"
  },
  "version_data": {
    "not_versioned": true,
    "default_version": "",
    "versions": {
      "v1": {
        "name": "v1"
      }
    }
  },
  "proxy": {
    "target_url": "http://httpbin.org",
    "strip_listen_path": true,
    "listen_path": "/hello-world/"
  }
}
"""
api_def = json.loads(api_def)

session = requests.Session()
session.headers.update({"Authorization": TYK_DASHBOARD_API_KEY})
create_response = session.post(TYK_DASHBOARD_API_URL + "/apis", json={"api_definition": api_def})
internal_api_id = create_response.json().get("Meta")
update_response = session.put(TYK_DASHBOARD_API_URL + "/apis/{}".format(internal_api_id),
                              json={"api_definition": api_def})

get_response = session.get(TYK_DASHBOARD_API_URL + "/apis/{}".format(internal_api_id))
api_id = get_response.json().get("api_definition").get("api_id")

# should print true if api_id is updated successfully, but prints false
print(api_id == api_def.get("api_id"))

```
**Actual behavior**
The provided `api_id` is ignored (when creating) / not updated (when updating)

**Expected behavior**
The `api_id` is changed to the provided value (when creating/updating a new api)

**Comment by christtyk** [comment]
Thanks for the report, we'll check it out!

**Comment by rewsmith** [comment]
To be clear, this is specifically an issue in SaaS/Hybrid. You can set the api_id with a PUT on-premise, but this does not work on a SaaS Dashboard.

**Comment by andyo-tyk** [comment]
This was fixed in Tyk 4.0.14, 5.0.3 and 5.1.1 releases.
